launcher/src/main/java/org/apache/spark/launcher/SparkLauncher.java

```diff
@@ -359,6 +364,82 @@ public SparkLauncher setVerbose(boolean verbose) {
   }
 
   /**
+   * Sets the working directory of the driver process.
```

Comment:
in the whole file: "driver" is misleading, since in cluster mode the child process is not the driver. I'd say "spark-submit" instead.